src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1048,6 +1095,17 @@ int sd_dhcp_server_forcerenew(sd_dhcp_server *server) {
         return r;
 }
 
+int sd_dhcp_server_set_bind_to_interface(sd_dhcp_server *server, int enabled) {
+        assert_return(server, -EINVAL);
+
+        if (enabled == server->bind_to_interface)
```

Comment:
so, this is supposed to be a public API, and in public APIs we so far exposed booleans as "int" for compt with pre-C99. internally we use "bool" everywhere however. This is mostly OK everywhere and the distinction doesn't matter much. But when comparing the two types we need to take special care since passing "2" in such a bool than is stored as "int" should be treated as true, but here it is treated as something that compares neither with true nor with false.

Hence, please, when comparing the parameter here, use `!!enabled == server->bind_to_interface`.